app/controllers/HomeController.java

```diff
@@ -140,4 +146,30 @@ public WebSocket ws() {
         });
     }
 
+    /**
+     * Checks that the WebSocket comes from the same origin.  This is necessary to protect
+     * against Cross-Site WebSocket Hijacking as WebSocket does not implement Same Origin Policy.
+     * <p>
+     * See https://tools.ietf.org/html/rfc6455#section-1.3 and
+     * http://blog.dewhurstsecurity.com/2013/08/30/security-testing-html5-websockets.html
+     */
+    public boolean sameOriginCheck(Http.RequestHeader rh) {
+        final String origin = rh.getHeader("Origin");
+
+        if (origin == null) {
+            logger.error("originCheck: rejecting request because no Origin header found");
+            return false;
+        } else if (originMatches(origin)) {
+            logger.debug("originCheck: originValue = " + origin);
+            return true;
+        } else {
+            logger.error("originCheck: rejecting request because Origin header value " + origin + " is not in the same origin");
+            return false;
+        }
+    }
+
+    private boolean originMatches(String origin) {
+        return origin.contains("localhost:9000") || origin.contains("localhost:19001");
+    }
```

Comment:
Great! We could package a helper in Akka HTTP, opened a ticket https://github.com/akka/akka/issues/20535
(Unrelated to this PR, but thanks for the reminder!)